pandas/io/sas/sas7bdat.py

```diff
@@ -703,15 +705,38 @@ def _chunk_to_dataframe(self):
                 rslt[name] = self._byte_chunk[jb, :].view(dtype=self.byte_order + "d")
                 rslt[name] = np.asarray(rslt[name], dtype=np.float64)
                 if self.convert_dates:
-                    unit = None
+                    warn_msg = (
```

Comment:
Does the user have some way to avoid this warning? Ideally, we would support all ov

1. warning when falling back to datetime.datetime (default)
2. Using datetime.datetime by default
3. Raising when we can't convert to datetime64

Reply:
Hi @TomAugspurger,
I can do this but I think it would require an extra parameter to the read_sas function to specify the date/datetime handling.  In a previous approach to fixing this issue I simply surfaced the underlying boolean convert_dates parameter from SAS7BDATReader up to pandas.read_sas, but @jreback preferred not to extend the number of parameters to read_sas.
I see 2 ways forward: 
1) add a new parameter to specify the date_handling format,
2) drop the warning and just return datetime64 when we can and datetime.datetime when we exceed the bounds of datetime64 (which is reasonably rare)

Which would you and Jeff prefer?

Reply:
Any thoughts on this @jreback? I slightly prefer an extra parameter to control the behavior of out-of-bounds timestamps, but could go either way.